hw/ip/otbn/dv/verilator/otbn_top_sim.cc

```diff
@@ -131,10 +132,15 @@ int main(int argc, char **argv) {
             << "==================" << std::endl
             << std::endl;
 
-  int ret_code = simctrl.Exec(argc, argv);
```

Comment:
Can't we continue to use `VerilatorSimCtrl::Exec` with it's behaviour fixed for `--help`?

Reply:
Well, the problem is that you need to "return" two things in that case: a return code plus a flag saying whether we actually ran a simulation. I guess we could change `VerilatorSimCtrl::Exec` to return a pair. Fortunately, the Ibex code that uses `VerilatorSimCtrl` already does this manually, so we can change the API without breaking it. Would you prefer that approach?

Reply:
It seems a bit silly to have a copy of the body of the `Exec` function repeated here, I think we should alter `Exec` so it works for this use case. Alternatively add a 'WasSimulationRun' function to `VerilatorSimCtrl` so the `Exec` function retains the same interface which is used to determine if registers should be dumped and the ISS/RTL comparison done.

Reply:
I much prefer the "more functional" style, so I've just pushed up a version that should look like that. What do you think?